packages/ilios-common/addon/components/course/collapsed-objectives.js

```diff
@@ -1,16 +1,15 @@
 import Component from '@glimmer/component';
-import { tracked } from '@glimmer/tracking';
-import { restartableTask } from 'ember-concurrency';
+import { cached } from '@glimmer/tracking';
+import { TrackedAsyncData } from 'ember-async-data';
 
 export default class CourseCollapsedObjectivesComponent extends Component {
-  @tracked objectivesRelationship;
-
-  load = restartableTask(async () => {
-    this.objectivesRelationship = await this.args.course.courseObjectives;
-  });
+  @cached
+  get objectivesData() {
+    return new TrackedAsyncData(this.args.course.courseObjectives);
+  }
 
   get objectives() {
-    return this.objectivesRelationship ? this.objectivesRelationship : [];
+    return this.objectivesData.isResolved ? this.objectivesData.value : null;
```

Comment:
the return value of this getter should remain to be an array/array-like, there are other getters sitting downstream from this that attempt to filter this output and would break on it (if they were to be invoked on empty data, which they currently are not).

having a consistent return type here also helps in streamlining conditional checks in the template.

Reply:
You are very right. I grabbed another example of this pattern and did not check to see what kind of return it was originally doing, so fixed.